app/controllers/admin/merchants_controller.rb

```diff
@@ -13,13 +13,19 @@ def edit
 
   def update
     merchant = Merchant.find(params[:id])
-    merchant.update(merchant_params)
-    flash[:success] = "#{merchant.name} has been successfully updated!"
-    redirect_to admin_merchant_path
+    if params[:status].present?
+      merchant.update_status(params[:status])
```

Comment:
I think I would've named this with a bang! since it's a destructive method, but it might be preferential. Might be worth asking in a review.